### PR TITLE
USVM-based type inference

### PR DESCRIPTION
**Pull request by Lipen**
This PR adds latest development related to USVM-based type inference support.

### CODE REVIEW

File: jacodb-ets/src/main/kotlin/org/jacodb/ets/dto/Convert.kt [hunk 1]
```diff
@@ -579,45 +615,62 @@ fun convertToEtsType(type: TypeDto): EtsType {
             }
         }
 
+        is AliasTypeDto -> EtsUnknownType // TODO: EtsAliasType
+        is AnnotationNamespaceTypeDto -> EtsUnknownType // TODO: EtsAnnotationNamespaceType
+        is AnnotationTypeQueryTypeDto -> EtsUnknownType // TODO: EtsAnnotationTypeQueryType
```
Comment (CaelmBleidd) [inline jacodb-ets/src/main/kotlin/org/jacodb/ets/dto/Convert.kt]: Lets create an issue with "all the things we don't know about" or unimplemented things for ArkTS and specify here a link on it

File: jacodb-ets/src/main/kotlin/org/jacodb/ets/graph/EtsApplicationGraph.kt [hunk 2]
```diff
@@ -42,13 +88,287 @@ class EtsApplicationGraphImpl(
...
+            get() = resolvedByCache + unresolvedByCache
+
+        val cacheMisses: Int
+            get() = total - cacheHits
+
+        private fun show(description: String, property: AtomicInteger): String {
+            if (total == 0) {
+                return "[N/A%] $description ${property.get()}"
+            }
+            val percent = 100 * property.get() / total
```
Comment (CaelmBleidd) [inline jacodb-ets/src/main/kotlin/org/jacodb/ets/graph/EtsApplicationGraph.kt]: Let's make this `percent` a double value , not integer

File: jacodb-ets/src/main/kotlin/org/jacodb/ets/graph/EtsApplicationGraph.kt [hunk 3]
```diff
@@ -42,13 +88,287 @@ class EtsApplicationGraphImpl(
...
+    private fun lookupClassWithIdealSignature(signature: EtsClassSignature): Maybe<EtsClass> {
+        require(signature.isIdeal())
+
+        if (signature in cacheClassWithIdealSignature) {
+            return cacheClassWithIdealSignature.getValue(signature)
+        }
+
+        val matched = cp.classes
+            .asSequence()
+            .filter { it.signature == signature && it.signature.file == signature.file }
```
Comment (CaelmBleidd) [inline jacodb-ets/src/main/kotlin/org/jacodb/ets/graph/EtsApplicationGraph.kt]: Does `signature.equals` take into account the file value?

File: jacodb-ets/src/main/kotlin/org/jacodb/ets/graph/EtsApplicationGraph.kt [hunk 4]
```diff
@@ -42,13 +88,287 @@ class EtsApplicationGraphImpl(
...
+        }
+
+        // NOTE: cache lookup MUST be performed AFTER trying to match the neighbour!
+        if (callee in cachePartiallyMatchedCallees) {
+            val s = cachePartiallyMatchedCallees.getValue(callee).asSequence()
+            if (s.none())
+                stats.cachedPartialMatchAsUnknown.incrementAndGet()
+            else
+                stats.resolvedByPartiallyMatchedCache.incrementAndGet()
+            return s
```
Comment (CaelmBleidd) [inline jacodb-ets/src/main/kotlin/org/jacodb/ets/graph/EtsApplicationGraph.kt]: Missing brackets

**Review by CaelmBleidd — CHANGES_REQUESTED** [review]

File: jacodb-ets/src/main/kotlin/org/jacodb/ets/dto/Convert.kt (hunk 1 above)
Comment (Lipen) [reply jacodb-ets/src/main/kotlin/org/jacodb/ets/dto/Convert.kt]: Added `EtsAliasType` and other types, mimicing the model from AA, though we are still not completely sure what they represent.

File: jacodb-ets/src/main/kotlin/org/jacodb/ets/graph/EtsApplicationGraph.kt (hunk 3 above)
Comment (Lipen) [reply jacodb-ets/src/main/kotlin/org/jacodb/ets/graph/EtsApplicationGraph.kt]: Yes, fixed.

File: jacodb-ets/src/main/kotlin/org/jacodb/ets/graph/EtsApplicationGraph.kt (hunk 2 above)
Comment (Lipen) [reply jacodb-ets/src/main/kotlin/org/jacodb/ets/graph/EtsApplicationGraph.kt]: In fact, probably this was made integer _intentionally_, to simplify its display by flooring it down to the integer. I've replaced it with a double value and use `percent.roundToInt()` when displaying.

File: jacodb-ets/src/main/kotlin/org/jacodb/ets/graph/EtsApplicationGraph.kt (hunk 4 above)
Comment (Lipen) [reply jacodb-ets/src/main/kotlin/org/jacodb/ets/graph/EtsApplicationGraph.kt]: Fixed. Fortunately, it was not me who introduced such a deviation from our blessed code style.

**Review by CaelmBleidd — APPROVED** [review]

**Pull request merged**